config/migrate/migrate.go

```diff
@@ -49,6 +61,67 @@ func ZeroToOne(cfg *config.Config) error {
 	}
 
 	cfg.Revision = 1
+
+	if err := safeWriteConfig(cfg); err != nil {
+		rollbackConfigWrite(cfg)
+		return err
+	}
+
+	return nil
+}
+
+// OneToTwo migrates a configuration from Revision 1 to Revision 2
+func OneToTwo(cfg *config.Config) error {
+
+	// TODO(ramfox): qfs migration
+
+	// TODO(arqu): config migration
+	// if err := oneToTwoConfig(cfg); err != nil {
+	// 	return err
+	// }
+	// TODO(arqu): config write/rollback
+	// if err := safeWriteConfig(cfg); err != nil {
+	// 	rollbackConfigWrite(cfg)
+	// 	return err
+	// }
+
+	// TODO(ramfox): remove original ipfs repo after all migrations were successful
+
+	return nil
+}
+
+func oneToTwoConfig(cfg *config.Config) error {
+	return nil
+}
+
+func rollbackConfigWrite(cfg *config.Config) {
+	cfgPath := cfg.Path()
+	if len(cfgPath) == 0 {
+		return
+	}
+	cfgPathComponents := strings.Split(cfgPath, "/")
+	cfgPathComponents[len(cfgPathComponents)-1] = "config_updated.yaml"
+	tmpCfgPath := strings.Join(cfgPathComponents, "/")
+	if _, err := os.Stat(tmpCfgPath); !os.IsNotExist(err) {
+		os.Remove(tmpCfgPath)
+	}
+}
+
+func safeWriteConfig(cfg *config.Config) error {
+	cfgPath := cfg.Path()
+	if len(cfgPath) == 0 {
+		return qerr.New(fmt.Errorf("invalid path"), "could not determine config path")
+	}
+	cfgPathComponents := strings.Split(cfgPath, "/")
+	cfgPathComponents[len(cfgPathComponents)-1] = "config_updated.yaml"
```

Comment:
Same thing here. Actually, if this is being done in two places, please extract into a helper function, such as `configUpdateFilename(cfgPath)`